product-service/src/functions/get-products-by-id/handler.ts

```diff
@@ -0,0 +1,56 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { formatJSONResponse } from '@libs/api-gateway';
+import { middyfy } from '@libs/lambda';
+
+import { Client } from 'pg';
+
+const {PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD} = process.env;
+
+const dbOptions = {
+  host: PG_HOST,
+  port: Number(PG_PORT),
+  database: PG_DATABASE, 
+  user: PG_USERNAME, 
+  password: PG_PASSWORD,
+  ssl: {
+    rejectUnauthorized: false
+  },
+  connectionTimeoutMillis: 5000
+};
+
+import schema from './schema';
+// import data from './mockproductlist';
+
+const getProductsById: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (event) => {
+
+  const client = new Client(dbOptions);
+  await client.connect();
+
+  try {
+    const productId = event['pathParameters']['productId'];
+    const { rows: item } = await client.query(`select p.id, s.count, p.price, p.title, p.description from products p left join stocks s on p.id = s.product_id where p.id = '${productId}';`);
+    
+    if (item) {
+      console.log('Get product: ', item)
+      return formatJSONResponse({
+        item,
+      });
+    } else {
+      console.log('Product not found')
+      return formatJSONResponse({
+        'result': 'Product not found',
+      });
+    }
+
+  } catch (err) {
+    console.error('Error: ', err);
+    return formatJSONResponse({
```

Comment:
formatJSONResponse always returns `statusCode` that equals to `200`. Whereas we have an optional requirement that says:
` All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)`

All public lambdas should meet this requirement in order to get optional point.